### ISSUE TITLE
The tensor size does not match when I am training a Chinese-English-Mixed model

### ISSUE THREAD

**Issue by busyyang** [myshell-ai/MeloTTS#248]
I tried to train a Chinese-English-Mixed model by our own dataset. The training precess worked, but with some   tensor size errors as below.

~~~
The expanded size of the tensor (32) must match the existing size (0) at non-singleton dimension 1.  Target sizes: [192, 32].  Tensor sizes: [192, 0]
~~~

And I noticed, there is a try .... catch part in training process.
~~~py
    ......
    for epoch in range(epoch_str, hps.train.epochs + 1):
        try:
            if rank == 0:
                train_and_evaluate(
                    rank,
                    epoch,
                    hps,
                    [net_g, net_d, net_dur_disc],
                    [optim_g, optim_d, optim_dur_disc],
                    [scheduler_g, scheduler_d, scheduler_dur_disc],
                    scaler,
                    [train_loader, eval_loader],
                    logger,
                    [writer, writer_eval],
                )
            else:
                train_and_evaluate(
                    rank,
                    epoch,
                    hps,
                    [net_g, net_d, net_dur_disc],
                    [optim_g, optim_d, optim_dur_disc],
                    [scheduler_g, scheduler_d, scheduler_dur_disc],
                    scaler,
                    [train_loader, None],
                    None,
                    None,
                )
        except Exception as e:
            print(e)
            torch.cuda.empty_cache()
        scheduler_g.step()
        scheduler_d.step()
        if net_dur_disc is not None:
            scheduler_dur_disc.step()
~~~

Why should we add the try ....catch code for errors, and why the size of tensor may be not match during the training?

**Comment by busyyang** [comment]
Because some audios is to short, it can be filted in TextAudioSpeakerLoader::_filter:
~~~py
class TextAudioSpeakerLoader(torch.utils.data.Dataset):
    """
    1) loads audio, speaker_id, text pairs
    2) normalizes text and converts them to sequences of integers
    3) computes spectrograms from audio files.
    """
    ....

    def _filter(self):
        """
        Filter text & store spec lengths
        """
        # Store spectrogram lengths for Bucketing
        # wav_length ~= file_size / (wav_channels * Bytes per dim) = file_size / (1 * 2)
        # spec_length = wav_length // hop_length

        audiopaths_sid_text_new = []
        lengths = []
        skipped = 0
        logger.info("Init dataset...")
        for item in tqdm(
            self.audiopaths_sid_text
        ):
            try:
                _id, spk, language, text, phones, tone, word2ph = item
            except:
                print(item)
                raise
            audiopath = f"{_id}"
            if self.min_text_len <= len(phones) and len(phones) <= self.max_text_len:
                spec_length = os.path.getsize(audiopath) // (2 * self.hop_length)
                if spec_length> self.segment_size:
                    phones = phones.split(" ")
                    tone = [int(i) for i in tone.split(" ")]
                    word2ph = [int(i) for i in word2ph.split(" ")]
                    audiopaths_sid_text_new.append(
                        [audiopath, spk, language, text, phones, tone, word2ph]
                    )
                    lengths.append(spec_length)
                else:
                    skipped += 1
            else:
                skipped += 1
        logger.info(f'min: {min(lengths)}; max: {max(lengths)}' )
        logger.info(
            "skipped: "
            + str(skipped)
            + ", total: "
            + str(len(self.audiopaths_sid_text))
        )
        self.audiopaths_sid_text = audiopaths_sid_text_new
        self.lengths = lengths
~~~

**Comment by Chengyang852** [comment]
请问训练后的效果怎么样，您使用了多少数据

**Comment by busyyang** [comment]
> 请问训练后的效果怎么样，您使用了多少数据

一个人大约20-30h数据训练出来效果还可以。
